### PR TITLE
Test: Like Store

### PR DESCRIPTION
**Pull request by hermannleboss**
## Description

Adding test to Like store to improve coverage

#284

### CODE REVIEW

File: src/pages/EntryPage.vue [hunk 1]
```diff
@@ -39,32 +39,30 @@ const shareStore = useShareStore()
...
-    await new Promise((resolve) => setTimeout(resolve, 2000)) // wait 2 seconds before continue
-  }
-
+//Handle 404
+myTimeout.value = setTimeout(() => {
   if (!entry.value?.id) {
     router.push('/404')
-    return
   }
+}, 30000)
```
Comment (stravid87) [inline src/pages/EntryPage.vue:54]: This is interesting code... why would there be up to a 3 sec delay before the  entry.value.id was populated?

File: src/pages/PromptPage.vue [hunk 2]
```diff
@@ -68,29 +69,25 @@ const prompt = computed(() => {
...
+const myTimeout = ref()
 
-onMounted(async () => {
-  await promptStore.fetchPrompts().catch((error) => errorStore.throwError(error))
-
-  if (!prompt.value?.id) {
-    await new Promise((resolve) => setTimeout(resolve, 2000)) // wait 2 seconds before continue
-  }
-
+//Handle 404
```
Comment (stravid87) [inline src/pages/PromptPage.vue:74]: The code here as well. What are the implications of these time outs?

File: src/stores/__tests__/likes.spec.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    await likeStore.addLike('entries', entyId)
+
+    // await waitUntil(() => {
+    //   return likeStore.getLikes?.length === initialLikeLenght + 1
+    // }).catch(() => console.log('1-First Like'))
+
+    expect(likeStore.getLikes?.length).toBe(initialLikeLenght + 1)
+    expect(likeStore.getDislikes?.length).toBe(initialDislikeLenght)
+
+    expect(likeStore.getLikes?.length).toMatchInlineSnapshot('1')
```
Comment (stravid87) [inline src/stores/__tests__/likes.spec.js:69]: Snaphsots! I like it!

**Review by stravid87 — COMMENTED** [review]
Great work adding all these tests: Vitest AND Cypress. The only thing that made me nervous was the use of time outs. We'll talk about it today.

File: src/pages/EntryPage.vue (hunk 1 above)
Comment (arnonrdp) [reply src/pages/EntryPage.vue]: This is 30 seconds.

**Comment by hermannleboss** [comment]
@arnonrdp this is ready for review

File: src/components/shared/ItemCard.vue [hunk 4]
```diff
@@ -43,20 +43,15 @@ const props = defineProps({
...
+  const wordCount = text.split(/\s+/).length
 
-  let wordCount = stringArr.length + adjustment
-  const averageWordsPerMinute = 238
+  // Calculate the reading time in minutes
+  const readingTimeInMinutes = wordCount / wordsPerMinute
 
-  return Math.round(wordCount / averageWordsPerMinute)
+  // Round up the reading time to the nearest integer
+  return Math.ceil(readingTimeInMinutes)
```
Comment (arnonrdp) [inline src/components/shared/ItemCard.vue:54]: There's nothing wrong with doing this.
This is actually excellent and very welcome!
But this type of refactoring makes the review a little longer as this is outside the scope of the PR.
Good job with that refactoring though.

**Review by arnonrdp — APPROVED** [review]
I'm approving your PR and will merge it, however I'm not sure the excessive amount of `onSnapshot` we're using is any good.
We are leaving several channels open with the Database just to have updates in real time.

Captura de Tela 2023-08-07 às 14 53 53

#### If possible, make sure that this cannot bring us any technical debt.

File: src/components/shared/ItemCard.vue (hunk 4 above)
Comment (hermannleboss) [reply src/components/shared/ItemCard.vue:54]: Yes, sorry, I understand, it's because I had several branches open and I didn't want to create a new one, I risked getting lost.
But i get it.

**Comment by hermannleboss** [comment]
> I'm approving your PR and will merge it, however I'm not sure the excessive amount of `onSnapshot` we're using is any good. We are leaving several channels open with the Database just to have updates in real time.
> 
> Captura de Tela 2023-08-07 às 14 53 53
> #### If possible, make sure that this cannot bring us any technical debt.

For the moment i don't see limitation of snapshot usage
